reana_job_controller/compute4punch_job_manager.py

```diff
@@ -0,0 +1,366 @@
+    """ regex to search the Job ID in a submit Proc line """
+
+    def __init__(
+        self,
+        docker_img=None,
+        cmd=None,
+        prettified_cmd=None,
+        env_vars=None,
+        workflow_uuid=None,
+        workflow_workspace=None,
+        cvmfs_mounts="false",
+        shared_file_system=False,
+        job_name=None,
+        c4p_cpu_cores=None,
+        c4p_memory_limit=None,
+        c4p_additional_requirements=None,
+        **kwargs,
+    ):
+        """
+        Compute4PUNCH Job Manager.
+
+        :param docker_img: Docker image.
+        :type docker_img: str
+        :param cmd: Command to execute.
+        :type cmd: list
+        :param prettified_cmd: prettified version of command to execute.
+        :type prettified_cmd: str
+        :param env_vars: Environment variables.
+        :type env_vars: dict
+        :param workflow_uuid: Unique workflow id.
+        :type workflow_uuid: str
+        :type workflow_workspace: str
+        :param cvmfs_mounts: list of CVMFS mounts as a string.
+        :type cvmfs_mounts: str
+        :param shared_file_system: if shared file system is available.
+        :type shared_file_system: bool
+        :param job_name: Name of the job
+        :type job_name: str
+        :param c4p_cpu_cores: number of CPU cores to use on C4P
+        :type c4p_cpu_cores: int, str
+        :param c4p_memory_limit: maximum memory to be used on C4P
+        :type c4p_memory_limit: int, str
+        :param c4p_additional_requirements: additional HTCondor requirements for the job
+        :type c4p_additional_requirements: str
+        """
+        super(Compute4PUNCHJobManager, self).__init__(
+            docker_img=docker_img,
+            cmd=cmd,
+            prettified_cmd=prettified_cmd,
+            env_vars=env_vars,
+            workflow_uuid=workflow_uuid,
+            workflow_workspace=workflow_workspace,
+            job_name=job_name,
+        )
+        self.c4p_connection = SSHClient(
+            hostname=C4P_LOGIN_NODE_HOSTNAME,
+            port=C4P_LOGIN_NODE_PORT,
+            timeout=C4P_SSH_TIMEOUT,
+            banner_timeout=C4P_SSH_BANNER_TIMEOUT,
+            auth_timeout=C4P_SSH_AUTH_TIMEOUT,
+            auth_strategy=motley_cue_auth_strategy_factory(
+                hostname=C4P_LOGIN_NODE_HOSTNAME
+            ),
+        )
+        self.compute_backend = "compute4punch"
+        self.cvmfs_mounts = cvmfs_mounts
+        self.shared_file_system = shared_file_system
+        self.job_execution_script_path = os.path.join(
+            self.c4p_abs_workspace_path, "run.sh"
+        )
+        self.job_description_path = os.path.join(
+            self.c4p_abs_workspace_path, "submit.jdl"
+        )
+
+        self.c4p_cpu_cores = c4p_cpu_cores or 8
+        self.c4p_memory_limit = c4p_memory_limit or 20000
+        self.c4p_additional_requirements = c4p_additional_requirements
+
+        self._workflow = None
```

Comment:
I think `_workflow` is never used

Reply:
Thanks, yes it not used at all. It is fixed in https://github.com/reanahub/reana-job-controller/pull/430/commits/97c63c9e42d5df43d5b542d62c60e33610141559.